src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const transformedState = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let nextState = { ...currentState };
```

Comment:
You don't need to declare new copy of state inside your iteration.
Instead you can provide copy to .push() as argument `.push({ ...currentState})`